### DISCUSSION TITLE
How could Pocketbase be integrated into a PHP stack just for the use of authentication?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#1736] [Q&A]
@ganigeorgiev Could you point me in the right direction of using Pocketbase as the central authority in an existing PHP application?  I'm trying to figure out how I could eventually move off of PHP in many stages.

I'm already using a middleware for setting a cookie during authentication so that I can use backend routes.  What I'm doing is that I have many PHP site silos with their own database.  It would be nice to be able to tie them into a Pocketbase central authority.

**Comment by ganigeorgiev** [comment]
I'm not really sure what to recommend and it will probably differ from one use case to another (and this is even more true considering that we don't have a PHP SDK yet).

I would expect the general flow to be something similar to the node SSR handling:

- if a token is sent with the request, validate it by making a call to the PocketBase server to refresh/fetch the PocketBase user
- perform your php operation and/or access checks
- if necessary update the cookie before returning the php response (_eg. if it is a sign-up or refresh request_)

**Reply by dschissler** [reply]
Does refreshing the PB user create a new token string?

Well, I'm far from needing to work on this, or really having a detailed idea how to integrate it.  It would really be a big deal for me though.  I'm thinking that it could be a pretty amazing way to upgrade a PHP system while having a Golang framework for slowly migrating code.  After getting auth going then an entire DB could be moved from Mysql/MariaDB into Pocketbase where it's called from PHP with just fixing bugs around the differences.  Then slowly views / screens could be switched over to the custom/framework Pocketbase.  It's a unique situation because the DB is also the application server.  Finally at the end a large refactor to ditch the PHP.

So I'm guessing that you'll be holding off on PHP work until you're ready to start playing with Presenter?  I'm not familiar with that software other than knowing that it's PHP and that it does some design organization thingy.

**Reply by ganigeorgiev** [reply]
> Does refreshing the PB user create a new token string?

When calling `POST /api/collections/users/auth-refresh` it will validate the token sent with the `Authorization` header and will return a new auth token with refreshed `exp` claim and up-to-date record data. The old token will still be valid until its `exp` time (_or until the user change their password or the `Record.TokenKey` is manually changed from some other action_).
